lightning/src/offers/signer.rs

```diff
@@ -393,19 +401,30 @@ fn verify_metadata<T: secp256k1::Signing>(
 			secp_ctx,
 			&SecretKey::from_slice(hmac.as_byte_array()).unwrap(),
 		);
-		if fixed_time_eq(&signing_pubkey.serialize(), &derived_keys.public_key().serialize()) {
+		#[allow(unused_mut)]
+		let mut ok = fixed_time_eq(&signing_pubkey.serialize(), &derived_keys.public_key().serialize());
```

Comment:
Is fixed_time_eq necessary here? I had to think of the onion failure parsing where we converted to a normal equals that the fuzzer knows how to work with.

Reply:
>Oh, no, I just meant that that rust-bitcoin PR makes fixed-time-eq a normal comparison when fuzzing.

But in that case, you could remove the &1==0, because the fuzzer finds its way? For the attr fail PR it wasn't necessary in the end either with a normal cmp.

>Taking up time doesn't fix it, someone can replay the exact same BOLT 12 message in a loop and measure the amount of time taken to validate it and slowly learn a correct signing pubkey. 

Got it. More subtle. Learning stuff here.